docs/specs/component.md

```diff
@@ -103,101 +104,139 @@ events:
   individual events. For example, emitting the `EventsReceived` event for 10
   events MUST increment the `component_received_events_total` counter by 10.
 
+#### ComponentBilled
+
+*Sources* MUST emit a `ComponentBilled` event that repsents the billing of
+enterprise customers.
+
+- Emission
+  - MUST emit only if the `enterprise.enabled` config option is set to `true`.
+  - MUST emit immediatley before the creation of Vector events and after
+    decompression and filtering of raw bytes.
+- Properties
+  - `byte_size` - REQUIRED, number of raw bytes received that will be turned
+    into Vector events.
+- Metrics
+  - MUST increment the `component_billed_bytes_total` counter by the defined
+    value with the defined properties as metric tags.
+- Logs
+  - MUST log a `Bytes billed.` message at the `trace` level with the defined
+    properties as key-value pairs.
+  - MUST NOT be rate limited.
+
 #### ComponentBytesReceived
 
-*Sources* MUST emit a `ComponentBytesReceived` event immediately after receiving, decompressing
-and filtering bytes from the upstream source and before the creation of a Vector event.
+*Sources* MUST emit a `ComponentBytesReceived` event that represents the
+reception of *raw event bytes*. It MUST emit immediately after receiving
+bytes from the upstream source and before decompression, filtering, and the
+creation of Vector events.
+
+- Emission
+  - MUST emit immediately after receiving bytes from the upstream source,
+    before decompression, filtering, and the creation of Vector events.
+- Properties
+  - `byte_size` - REQUIRED, number of raw event bytes received.
+    - For HTTP-based protocols, the total number of raw bytes in the HTTP body,
+      as typically represented by the `Content-Length` header.
+    - For UDP, TCP, and Unix protocols, the total number of raw bytes received
+      from the socket excluding the delimiter.
+    - For files, the total number of raw bytes read from the file excluding the
+      delimiter.
+  - `protocol` - REQUIRED, the protocol used to send the bytes. MUST be one of
+    `tcp`, `udp`, `unix`, `http`, `https`, `file`, etc.)
+  - `http_path` - OPTIONAL, the HTTP path, excluding query strings.
+  - `socket` - OPTIONAL, the socket number that bytes were received from.
+- Metrics
+  - MUST increment the `component_received_bytes_total` counter by the defined
+    value with the defined properties as metric tags.
+- Logs
+  - MUST log a `Bytes received.` message at the `trace` level with the defined
+    properties as key-value pairs.
+  - MUST NOT be rate limited.
+
+#### ComponentBytesSent
+
+*Sinks* MUST emit a `ComponentBytesSent` that represents the number of raw
+bytes sent downstream.
 
+- Emission
+  - MUST emit immediately after *successful* transmission the bytes.
+  - MUST emit before compression.
+  - MUST NOT emit for pull-based sinks since they do not send data. For
+    example, the `prometheus_exporter` sink MUST NOT emit this event.
```

Comment:
Based on the naming, as a user, I'd expect this to be the inverse of `ComponentBytesReceived`. That is: it would indicate the raw bytes sent by the component over, for example, a TCP socket. This would mean both successful and unsuccessful requests. It would also mean pull based sinks would emit this metric.

Reply:
This does not include unsuccessful requests, it represents the successful delivery of bytes. This is an internal component metric, not a host-level network metric.